src/Faker/Generator.php

```diff
@@ -205,6 +205,55 @@ class Generator
     protected $providers = [];
     protected $formatters = [];
 
+    /**
+     * @var UniqueGenerator
+     */
+    private $uniqueGenerator;
+
+    public function __construct()
+    {
+        $this->uniqueGenerator = new UniqueGenerator($this);
+    }
+
+    /**
+     * With the unique generator you are guaranteed to never get the same two
+     * values.
+     *
+     * @return self The UniqueGenerator is a proxy
+     */
+    public function withUnique()
+    {
+        return $this->uniqueGenerator;
+    }
+
+    /**
+     * Get a value only some percentage of the time.
+     *
+     * @param float $weight A probability between 0 and 1, 0 means that we always get the default value.
+     *
+     * @return self
+     */
+    public function withMaybe(float $weight = 0.5, $default = null)
```

Comment:
```suggestion
    public function withMaybe(float $weight = 0.5, $default = null): self
```

Reply:
What about `chance`?

Reply:
I've added `chance()` and deprecated `optional()`. I also deprecated `DefaultGenerator`.